Source/Reporters/JunitXMLReporter.jsl

```diff
@@ -0,0 +1,247 @@
+﻿/*
+	Class: UtJunitXMLReporter
+		---Prototype---
+		class UtJunitXMLReporter inherits UtReporter
+		---------------
+		Collect failure information about test cases.
+		Then issue a report in JUnitXML format on demand.
+		
+		Example of JunitXML format:
+		
+		<?xml version="1.0" encoding="UTF-8" ?> 
+		   <testsuites id="20140612_170519" name="New_configuration (14/06/12 17:05:19)" tests="225" failures="1262" time="0.001">
+			  <testsuite id="codereview.cobol.analysisProvider" name="COBOL Code Review" tests="45" failures="17" time="0.001">
+				 <testcase id="codereview.cobol.rules.ProgramIdRule" name="Use a program name that matches the source file name" time="0.001">
+					<failure message="PROGRAM.cbl:2 Use a program name that matches the source file name" type="WARNING">
+		WARNING: Use a program name that matches the source file name
+		Category: COBOL Code Review – Naming Conventions
+		File: /project/PROGRAM.cbl
+		Line: 2
+			  </failure>
+			</testcase>
+		  </testsuite>
+		</testsuites>
+*/
+
+Define Class(
+	"UtJunitXMLReporter",
+	Base Class( UtCollectingReporter ),
+
+/*	Function: generate_xml_tag
+        ---Prototype---
+        generate_xml_tag(<tag>, <content>, <properties>)
+        ---------------
+        takes in a tag as a string and content, puts the content into the tag in xml format
+      
+    Arguments:
+		tag: string: string that the tag should be
+		content: string: content that goes inside the string
+		properties: associative array: associative array with properties as keys and the values
+
+    Example:
+        This is an example
+        ---JSL---
+		test = generate_xml_tag("test", "content", ["prop"=>1]);
+
+		Expected Return:
+		<test prop="1">
+		  content
+		</test>
+        ---------
+*/
+	generate_xml_tag = Method({tag, content, properties},
+		xml_tag = "<" || tag;
+		props = properties << get keys;
+		vals = properties << get values;
+		For(i=1, i<=N Items(props), i++,
+			name = props[i];
+			If(Type(name) != "String",
+				Throw("property argument has a non string key found type:" || Type(name));
+			);
+			xml_tag ||= " " || name || "=\!"" || Char(vals[i]) || "\!"";
+		);
+		xml_tag ||= ">" || "\!n";
+		
+		If(content != "",
+			// replace all newline with doublespace then newline
+			Substitute Into(content, "\!n", "\!n  ");
+			xml_tag ||= "  " || content || "\!n";
```

Comment:
`vals` and `content` often both come from human-readable free-form text. It may be part of an expected or actual value. It may be written by a matcher. It may be written by a user as a test case name, test name, or assertion label. Or something else. Please use `XML Encode` for attribute values and XML text to ensure they don't contain invalid characters or prematurely end the attribute or node.

Reply:
I added some failing tests in https://github.com/sassoftware/jsl-hamcrest/pull/60/commits/6346b9fe8356a2e990bf91c6c2a7068106aa0dea for this.

Reply:
fixed to escape non valid xml characters in commit d282dd545646455c25c93815c68331cc840b0e6d